### ISSUE TITLE
custom schema

### ISSUE THREAD

**Issue by krichards** [romeerez/orchid-orm#156]
I'm trying to use a custom schema, but struggling to make it work.

If I just add the schema to the database config e.g.

```
 const db = orchidORM({
  ...
    schema: 'oidc',
}....
```

When I try and run the migration it barfs with 
```
error: no schema has been selected to create in
    at /home/kevin/projects/haven/haven-identity/node_modules/.pnpm/pg@8.11.2/node_modules/pg/lib/client.js:526:17
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async performQuery$1 (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/pqb@0.16.5/node_modules/pqb/dist/index.mjs:3826:12)
    at async createSchemaMigrations (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:153:5)
    at async getMigratedVersionsMap (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:2135:7)
    at async migrateOrRollback (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:2185:30)
    at async runCommand (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:3718:7)

```

Which presumably is because the migration version table is trying to be created and the schema 'oidc' does not yet exist?
This error happens before any migration is attempted.

If I create the schema manually then I get a different error trying to migrate, and checking in the DB - the schemaMigrations table has been created in the custom schema.

```
error: duplicate key value violates unique constraint "pg_type_typname_nsp_index"
    at /home/kevin/projects/haven/haven-identity/node_modules/.pnpm/pg@8.11.2/node_modules/pg/lib/client.js:526:17
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async performQuery$1 (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/pqb@0.16.5/node_modules/pqb/dist/index.mjs:3826:12)
    at async createSchemaMigrations (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:153:5)
    at async getMigratedVersionsMap (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:2135:7)
    at async migrateOrRollback (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:2185:30)
    at async runCommand (file:///home/kevin/projects/haven/haven-identity/node_modules/.pnpm/rake-db@2.10.9/node_modules/rake-db/dist/index.mjs:3718:7) {
  length: 257,
  severity: 'ERROR',
  code: '23505',
  detail: 'Key (typname, typnamespace)=(schemaMigrations, 16387) already exists.',
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: 'pg_catalog',
  table: 'pg_type',
  column: undefined,
  dataType: undefined,
  constraint: 'pg_type_typname_nsp_index',
  file: 'nbtinsert.c',
  line: '663',
  routine: '_bt_check_unique'
}
```

What am I missing?

**Comment by romeerez** [comment]
Thank you for filing an issue!

First thing, you've added `schema: 'oidc',` to the `orchidORM` config. But that's for ORM, and migrations don't use that `const db = orchidORM(...)`. Did you add the same schema into `rakeDb` config?

The schema can be set to the `DATABASE_URL` in the .env file:
```ts
postgres://user:password@localhost:5432/dbname?schema=custom
```
And then you provide this `DATABASE_URL` to both `orchidORM` and `rakeDb` and they both will use the same db schema.

I checked `db create` - it works, created a new db with a schema and `schemaMigrations` table in it. `db drop` and `db reset` also works. If I try `db migrate` without creating a db, it throws expectedly `database "db" does not exist`.

And when I drop the db, create it manually with `psql`, and then run `db migrate` I have the same error as you reported.

To solve the first problem, run `db create`: it will skip creating db if it already exists, then it will create a schema and the migrations table, and after it `db migrate` will work. I'll update the logic of `db migrate` for it to create the schema if it doesn't exist automatically, so users won't have to worry about it.

The second problem:
> If I create the schema manually then I get a different error trying to migrate, and checking in the DB - the schemaMigrations table has been created in the custom schema.

The error is weird, I tried but can't reproduce it.
